examples/c/gpio_advanced.c

```diff
@@ -16,51 +16,100 @@
 
 /* mraa header */
 #include "mraa/gpio.h"
+#include "mraa/led.h"
+
+/* GPIO default */
+#define GPIO_PIN1 5
 
-#define GPIO_PIN 6
+/* LED default */
+#define LED_NUM1 0
 
-void
-int_handler(void* args)
-{
+volatile int isr_occurred;
+
+void int_handler() {
     fprintf(stdout, "ISR triggered\n");
+    isr_occurred = 1;
 }
 
-int
-main()
-{
+void isr_do(mraa_led_context led1) {
+    /* turn LED on/off 10 times depending on max_brightness value */
+    for(int i =0; i < 10; i++) {
+        mraa_led_set_brightness(led1, 1);
+        usleep(50000);
+        mraa_led_set_brightness(led1, 0);
+        usleep(50000);
+    }
+}
+
+
+int main(int argc, char *argv[]) {
     mraa_result_t status = MRAA_SUCCESS;
     mraa_gpio_context gpio;
+    mraa_led_context led1;
+    int gpio_pin1, led_num1;
+    gpio_pin1 = (argc >= 2)?atoi(argv[1]):GPIO_PIN1;
+    led_num1 = (argc >= 3)?atoi(argv[2]):LED_NUM1;
 
+    
     /* initialize mraa for the platform (not needed most of the times) */
     mraa_init();
 
     //! [Interesting]
     /* initialize GPIO pin */
-    gpio = mraa_gpio_init(GPIO_PIN);
+    gpio = mraa_gpio_init(GPIO_PIN1);
+    /* Wait for the GPIO to initialize */
+    usleep(5000);
     if (gpio == NULL) {
-        fprintf(stderr, "Failed to initialize GPIO %d\n", GPIO_PIN);
+        fprintf(stderr, "Failed to initialize GPIO %d\n", GPIO_PIN1);
         mraa_deinit();
         return EXIT_FAILURE;
     }
 
-    /* set GPIO to input */
+    /* initialize LED */
+    led1 = mraa_led_init(led_num1);
+    usleep(5000);;
+    if (led1 == NULL) {
+        fprintf(stderr, "Failed to initialize LED 1\n");
+        mraa_deinit();
+        return EXIT_FAILURE;
+    }
+
+    /* set GPIO1 to input */
     status = mraa_gpio_dir(gpio, MRAA_GPIO_IN);
+    usleep(5000);
     if (status != MRAA_SUCCESS) {
+        fprintf(stderr, "Failed to set GPIO1 to input\n");
         goto err_exit;
     }
 
     /* configure ISR for GPIO */
-    status = mraa_gpio_isr(gpio, MRAA_GPIO_EDGE_BOTH, &int_handler, NULL);
+    status = mraa_gpio_isr(gpio, MRAA_GPIO_EDGE_RISING, &int_handler, NULL);
+    /* Wait for the ISR to configure */
+    usleep(5000);
     if (status != MRAA_SUCCESS) {
+        fprintf(stderr, "Failed to configure ISR\n");
         goto err_exit;
     }
+    
+    int c = 0;
```

Comment:
I prefer this:
```c
while (1) {
  if (isr_occurred) {
    isr_occrred = 0;
    isr_do(led1);
    break;
  }
  usleep(5000);
}
```

Reply:
In fact, I think your example can be as simple as Python example.
Since this is only an example for customers to know how to use GPIO ISR, no need to be too complicated.

Reply:
OK